### PR TITLE
IGNITE-12892 Clarify WAL archive size configuration

### PR DESCRIPTION
**Pull request by SammyVimes**
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/configuration/DataStorageConfiguration.java [hunk 1]
```diff
@@ -594,21 +597,29 @@ public boolean isWalHistorySizeParameterUsed() {
     /**
      * Gets a max allowed size(in bytes) of WAL archives.
      *
-     * @return max size(in bytes) of WAL archive directory(always greater than 0).
+     * @return max size(in bytes) of WAL archive directory(greater than 0, or {@link #UNLIMITED_WAL_ARCHIVE} if
+     * WAL archive size is unlimited).
      */
     public long getMaxWalArchiveSize() {
+        if (maxWalArchiveSize == UNLIMITED_WAL_ARCHIVE)
```
Comment (tkalkirill) [inline modules/core/src/main/java/org/apache/ignite/configuration/DataStorageConfiguration.java:604]: It seems that only the field value should be returned here, and the invariants should be checked in the setter.

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/checkpoint/CheckpointHistory.java [hunk 2]
```diff
@@ -67,8 +67,11 @@
     /** The maximal number of checkpoints hold in memory. */
     private final int maxCpHistMemSize;
 
-    /** If WalHistorySize was setted by user will use old way for removing checkpoints. */
-    private final boolean isWalHistorySizeParameterEnabled;
+    /** Should WAL be truncated */
+    private final boolean isWalTruncationEnabled;
+
+    /** Maximum size of WAL archive (in bytes) */
+    private final long maxWalArchiveSize;
```
Comment (tkalkirill) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/checkpoint/CheckpointHistory.java:74]: It seems that this field is not used anywhere other than the constructor.

**Comment by tkalkirill** [comment]
LGTM

**Comment by bmv126** [comment]
@SammyVimes  / @tkalkirill 
We were facing issue mentioned in http://apache-ignite-users.70518.x6.nabble.com/Could-not-clear-historyMap-due-to-WAL-reservation-on-cp-td34779.html
Now I see that as part of this PR, it has modified code changes wrt to the warning seen.

Does this PR solve this issue ?

**Comment by tkalkirill** [comment]
Hi, @bmv126 !

This PR does not fix the problem described in the discussion above. As far as I understand, the problem is as follows:
Segment with identifier 1468 was reserved, therefore it and all subsequent ones cannot be deleted until they are released.
As far as I know, segments can be reserved for historical rebalancing. I looked at the code and found that segments can be reserved for PageHistoryDiagnoster. Maybe bug here. Perhaps this can be understood by the log for the PageHistoryDiagnoster.

**Pull request closed without merging**